### ISSUE TITLE
Improve XGBoost Model

### ISSUE THREAD

**Issue by appleroll** [appleroll-research/promptforest#3]
The current XGBoost has not been tested with extra hyperparameters. Moreover, it trains on only 260k samples which should be enough, but could be increased if we are to remove all test samples in the training data (the current training data is contaminated with test samples). 

One dataset could be https://huggingface.co/datasets/m4vic/prompt-injection-dataset

This issue is pretty big, so my request is for pull requests that work towards this problem, no matter how small or how big. Whether it be rewriting the Jupyter Notebook so it doesn't use test samples, or adding adding hyperparameter sweeps.

**Comment by dtee1** [comment]
Hello, I'm interested please assign to me.

**Comment by megha2720** [comment]
Hi @appleroll, I see this issue was assigned a while ago but is still open. If it's no longer being worked on, I'd love to take a crack at it! Could you please assign it to me?

**Comment by appleroll** [comment]
Sure!

**Comment by megha2720** [comment]
Hi @appleroll, I've submitted the PR for this issue (#7). I removed the duplicate entries and tuned the hyperparameters to reach 92.73% accuracy. Ready for your review!
